modules/twinklebatchundelete.js

```diff
@@ -122,33 +144,71 @@ Twinkle.batchundelete.callback.evaluate = function( event ) {
 		return;
 	}
 
-	var batchOperation = new Morebits.batchOperation("Undeleting pages");
-	batchOperation.setOption("chunkSize", Twinkle.getPref('batchUndeleteChunks'));
-	batchOperation.setOption("preserveIndividualStatusLines", true);
-	batchOperation.setPageList(pages);
-	batchOperation.run(function(pageName) {
-		var query = {
-			'token': mw.user.tokens.get().editToken,
-			'title': pageName,
-			'action': 'undelete',
-			'reason': reason + Twinkle.getPref('deletionSummaryAd'),
-			'undel_talk': undel_talk	// Not used in the API request, but added here for access in onSuccess()
+
+	var pageUndeleter = new Morebits.batchOperation("Undeleting pages");
+	pageUndeleter.setOption("chunkSize", Twinkle.getPref('batchUndeleteChunks'));
+	pageUndeleter.setOption("preserveIndividualStatusLines", true);
+	pageUndeleter.setPageList(pages);
+	pageUndeleter.run(function(pageName) {
+		var params = {
+			page: pageName,
+			undel_talk: undel_talk,	// Not used in the API request, but added here for access in onSuccess()
```

Comment:
```suggestion
			undel_talk: undel_talk,
```
No need for that comment now, as this is now in `params`, not in `query`